experimental/grpc_attestation_client/java/com/google/oak/grpc_attestation_client/AttestationClient.java

```diff
@@ -0,0 +1,118 @@
+package com.google.oak.grpc_attestation_client;
+
+import com.google.oak.grpc_attestation_client.AeadEncryptor;
+import com.google.oak.grpc_attestation_client.KeyNegotiator;
+import com.google.protobuf.ByteString;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+import java.lang.RuntimeException;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import oak.examples.grpc_attestation.AttestedInvokeResponse;
+import oak.examples.grpc_attestation.AttestedInvokeRequest;
+import oak.examples.grpc_attestation.ClientIdentity;
+import oak.examples.grpc_attestation.ServerIdentity;
+import oak.examples.grpc_attestation.GrpcAttestationGrpc;
+import oak.examples.grpc_attestation.GrpcAttestationGrpc.GrpcAttestationStub;
+
+public class AttestationClient {
```

Comment:
This works fine for the gRPC case, but it is too tied to the underlying transport. I think we should still have a separate state machine that accepts messages or Java objects to move from one state to the next, and that may be used in any context (e.g. it may be used on a server, since the protocol is symmetric). Then of course a wrapper like this that is tied to gRPC would be trivial to implement on top of that.

Edit: perhaps KeyNegotiation is similar to what I'm describing already, but it should itself hand out the encryptor / decryptor in a way that makes it impossible to make mistakes. Anyways, we can do this separately.

Reply:
Moved encryptor creation to `KeyNegotiator`.
Also created an issue for implementing a protocol independent state machine: https://github.com/project-oak/oak/issues/2121